liquid-rust-syntax/src/surface.rs

```diff
@@ -260,6 +302,260 @@ impl Desugar {
     }
 }
 
-pub fn desugar(ssig: FnSig) -> ast::FnSig {
+pub fn desugar(ssig: BareFnSig) -> ast::FnSig {
     Desugar::desugar(ssig)
 }
+
+// ---------------------------------------------------------------------------
+// -------------------------- DEFAULT Signatures -----------------------------
+// ---------------------------------------------------------------------------
+
+fn default_refkind(m: &Mutability) -> RefKind {
+    match m {
+        Mutability::Mut => RefKind::Mut,
+        Mutability::Not => RefKind::Immut,
+    }
+}
+
+fn kind_def_ident(k: &rustc_middle::ty::TyKind) -> DefIdent {
+    match k {
+        rustc_middle::ty::TyKind::Bool => Layout::Bool,
+        rustc_middle::ty::TyKind::Int(i) => Layout::Int(*i),
+        rustc_middle::ty::TyKind::Uint(u) => Layout::Uint(*u),
+        rustc_middle::ty::TyKind::Float(f) => Layout::Float(*f),
+        rustc_middle::ty::TyKind::Adt(adt, _) => Layout::Adt(adt.did),
+        _ => panic!("kind_def_ident  : {:?}", k),
+    }
+}
+
+fn default_base_path(k: &rustc_middle::ty::TyKind, span: Span) -> DefPath {
+    Path { ident: kind_def_ident(k), args: None, span }
+}
+
+fn default_path(k: &rustc_middle::ty::TyKind, span: Span) -> DefPath {
+    match k {
+        rustc_middle::ty::TyKind::Bool => default_base_path(k, span),
+        rustc_middle::ty::TyKind::Int(_) => default_base_path(k, span),
+        rustc_middle::ty::TyKind::Uint(_) => default_base_path(k, span),
+        rustc_middle::ty::TyKind::Float(_) => default_base_path(k, span),
+        rustc_middle::ty::TyKind::Adt(_, args) => {
+            let ts = args.types().map(|arg| default_ty(&arg, span)).collect();
+            Path { ident: kind_def_ident(k), args: Some(ts), span }
+        }
+
+        _ => panic!("default_path fails on: {:?} at {:?}", k, span),
+    }
+}
+
+fn default_ty_kind(k: &rustc_middle::ty::TyKind, span: Span) -> DefTyKind {
+    match k {
+        rustc_middle::ty::TyKind::Ref(_, ty, m) => {
+            let ref_kind = default_refkind(m);
+            let tgt_ty = default_ty(ty, span);
+            TyKind::Ref(ref_kind, Box::new(tgt_ty))
+        }
+        rustc_middle::ty::TyKind::Param(a) => TyKind::Param(*a),
+        _ => TyKind::Base(default_path(k, span)),
+    }
+}
+fn default_ty(t: &rustc_middle::ty::Ty, span: Span) -> DefTy {
+    let kind = default_ty_kind(t.kind(), span);
+    Ty { kind, span }
+}
+
+fn mk_ident(i: i32, span: Span) -> Ident {
+    let xstr = format!("def_x_{}", i);
+    Ident::from_str_and_span(&xstr, span)
+}
+
+pub fn default_fn_sig(rust_sig: rustc_middle::ty::FnSig, span: Span) -> DefFnSig {
```

Comment:
TODO

We probably won't need the `Span` if we have `ResolvePath` and `UnresolvedPath`